### ISSUE TITLE
form doesnt submit multiple times and textfield are not getting reset

### ISSUE THREAD

**Issue by rodriger06** [GiancarloCode/form_bloc#107]
I used some of the code found in https://giancarlocode.github.io/form_bloc/#/  to create my form.
when my form is submitted, instead of calling 
Navigator.of(context).pushReplacement(
             MaterialPageRoute(builder: (_) => SuccessScreen()));
and displaying another screen with success message, i am calling  a snack bar to display at the bottom of the screen
Scaffold.of(context).showSnackBar(
                      SnackBar(content: Text(state.successResponse)));

when i submit my form the first time, the submission works.  i get the snackbar message. however, i want to enter more data in the same form.  the second time i submit, then the form doesnt work anymore and cannot submit.  also i notice the my textfields are not getting reset to blank.

i have the following code to reset the fields

  final amount = TextFieldBloc(
    validators: [
      FieldBlocValidators.required,
    ],
  );

  final frequency = TextFieldBloc(
    validators: [
      FieldBlocValidators.required,
    ],
  );

  clearTextField() {
    amount.clear();
    frequency.clear();
  }

can you kindly help resolve these issues?  how can i reset the textfields when form is submitted and be able to enter new data to submit again in the same form?

**Comment by GiancarloCode** [comment]
Hello, use `canSubmitAgain` but default is false, for forms like login.
and call `clear()`;

```dart
  emitSuccess(canSubmitAgain: true);
  clear();
```

**Comment by rodriger06** [comment]
thank you for your help.    emitSuccess(canSubmitAgain: true);  works and i can resubmit the form. however, clear() is not working and the textfields still have the data that was submitted.

**Comment by rodriger06** [comment]
thanks it looks likes it is working now.  i  have a question regarding calling the submit function.  in my form, i have the following code

 Expanded(
                              child: RaisedButton(
                                child: Text(save_and_continue,
                                    style: TextStyle(color: white, fontSize: margin_16dp)),
                                color: colorPrimary,
                                **onPressed: addEditIncomeBloc.submit**  
                ),
   ),

if you notice , the submit call is on one line I marked in bold.  this works fine but if i want to call other functions after pressing the button, then the submit doesnt work.  for example, if i change the code to the following:
 **onPressed: () { 
   addEditIncomeBloc.submit;
    otherFunction();
}**  
this code doesnt work and the submit doesnt get called.  it looks like submit only get called is onpressed is one liner calling submit only.  also when I added the {} , android studio give me a warning;  "avoid unnecessary statements"

how can i fix this so that i can call submit and any other function i defined in my class?
